flake.nix

```diff
@@ -26,6 +26,7 @@
                 system.nixos.versionSuffix =
                   ".${final.substring 0 8 (self.lastModifiedDate or self.lastModified or "19700101")}.${self.shortRev or "dirty"}";
                 system.nixos.revision = final.mkIf (self ? rev) self.rev;
+                nix.registry.nixos.flake = "${self}";
```

Comment:
We're supposed to be the distro with hermeticity and no mutable global state magic. The registry can't be fixed. Do not use it.

```suggestion
```

Reply:
This would be my response: https://github.com/NixOS/nix/issues/7422#issuecomment-1341367725